code/modules/ruins/spaceruin_code/hilbertshotel.dm

```diff
@@ -0,0 +1,343 @@
+    storageTurf = GLOB.hhStorageTurf
+
+/obj/item/hilbertshotel/attack_self(mob/user)
+    . = ..()
+    var/chosenRoomNumber = input("What number room will you be checking into?", "Room Number") as null|num
+    if(!chosenRoomNumber)
+        return
+    if((chosenRoomNumber < 1) || (chosenRoomNumber != round(chosenRoomNumber)))
+        to_chat(user, "<span class='warning'>That is not a valid room number!</span>")
+        return
+    src.forceMove(get_turf(user))
+    if(tryActiveRoom(chosenRoomNumber, user))
+        return
+    if(tryStoredRoom(chosenRoomNumber, user))
+        return
+    sendToNewRoom(chosenRoomNumber, user)
+
+/obj/item/hilbertshotel/proc/tryActiveRoom(var/roomNumber, var/mob/user)
+    if(activeRooms["[roomNumber]"])
+        var/datum/turf_reservation/roomReservation = activeRooms["[roomNumber]"]
+        do_sparks(3, FALSE, get_turf(user))
+        user.forceMove(locate(roomReservation.bottom_left_coords[1] + hotelRoomTemp.landingZoneRelativeX, roomReservation.bottom_left_coords[2] + hotelRoomTemp.landingZoneRelativeY, roomReservation.bottom_left_coords[3]))
+        return TRUE
+    else
+        return FALSE
+
+/obj/item/hilbertshotel/proc/tryStoredRoom(var/roomNumber, var/mob/user)
+    if(storedRooms["[roomNumber]"])
+        var/datum/turf_reservation/roomReservation = SSmapping.RequestBlockReservation(hotelRoomTemp.width, hotelRoomTemp.height)
+        hotelRoomTempEmpty.load(locate(roomReservation.bottom_left_coords[1], roomReservation.bottom_left_coords[2], roomReservation.bottom_left_coords[3]))
+        var/turfNumber = 1
+        for(var/i=0, i<hotelRoomTemp.width, i++)
+            for(var/j=0, j<hotelRoomTemp.height, j++)
+                for(var/atom/movable/A in storedRooms["[roomNumber]"][turfNumber])
+                    if(ismob(A))
+                        var/mob/M = A
+                        M.notransform = FALSE
+                    A.forceMove(locate(roomReservation.bottom_left_coords[1] + i, roomReservation.bottom_left_coords[2] + j, roomReservation.bottom_left_coords[3]))
+                turfNumber++
+        for(var/obj/abstracthotelstorage/S in storageTurf)
+            if(S.roomNumber == roomNumber)
+                qdel(S)
+        storedRooms -= "[roomNumber]"
+        activeRooms["[roomNumber]"] = roomReservation
+        linkTurfs(roomReservation, roomNumber)
+        do_sparks(3, FALSE, get_turf(user))
+        user.forceMove(locate(roomReservation.bottom_left_coords[1] + hotelRoomTemp.landingZoneRelativeX, roomReservation.bottom_left_coords[2] + hotelRoomTemp.landingZoneRelativeY, roomReservation.bottom_left_coords[3]))
+        return TRUE
+    else
+        return FALSE
+
+/obj/item/hilbertshotel/proc/sendToNewRoom(var/roomNumber, var/mob/user)
+    var/datum/turf_reservation/roomReservation = SSmapping.RequestBlockReservation(hotelRoomTemp.width, hotelRoomTemp.height)
+    if(ruinSpawned && (roomNumber == 1337256))
```

Comment:
1. Why a hardcoded number?
2. Why this number?
3. Aren't numbers in byond 16-bit, thus numbers outside [-65536,65535] are actually floats?

Reply:
It's a part of the lore. As for how the number is handled, its handled fine.

Reply:
I noticed that bit of lore/puzzle.
This might be a good place for a comment.